t/14-grutasks-git.t

```diff
@@ -448,14 +452,102 @@ subtest 'delete_needles' => sub {
+    };
+
+    subtest 'file present and older => fail' => sub {
+        my $mock_file = $tmpdir->child('mock.gitlab.flag')->touch;
+        my $old_time = time - 2000;
+        utime($old_time, $old_time, $mock_file)
+          or die "Couldn't change mtime on file: $mock_file - $!";
+        my $outcome = report_server_unavailable($t->app, 'gitlab');
+        is $outcome, 'FAIL', 'Fails the job when mtime >= 1800';
+    };
+
+    subtest 'report_server_available removes existing flag file' => sub {
+        my $mock_file = $tmpdir->child('mock.gitlab.flag')->touch;
+        ok -f $mock_file, 'mock.gitlab.flag exists initially';
+        report_server_available($t->app, 'gitlab');
+        ok !-f $mock_file, 'gitlab flag file was removed';
+    };
+
+    subtest 'multiple servers do not conflict' => sub {
+        my $outcome_gitlab = report_server_unavailable($t->app, 'gitlab');
+        is $outcome_gitlab, 'SKIP', 'No gitlab flag => skip';
+        my $gitlab_file = "$tmpdir/mock.gitlab.flag";
+        ok -f $gitlab_file, 'Created mock.gitlab.flag';
+        my $outcome_github = report_server_unavailable($t->app, 'github');
+        is $outcome_github, 'SKIP', 'No github flag => skip';
+        my $github_file = "$tmpdir/mock.github.flag";
+        ok -f $github_file, 'Created mock.github.flag';
+
+        my $old_time = time - 2000;
+        utime($old_time, $old_time, $github_file);
+        # Re-check
+        $outcome_gitlab = report_server_unavailable($t->app, 'gitlab');
+        $outcome_github = report_server_unavailable($t->app, 'github');
+        is $outcome_gitlab, 'SKIP', 'GitLab file still fresh => skip';
+        is $outcome_github, 'FAIL', 'GitHub file older => fail';
+    };
+
+    subtest Clone => sub {
+        my $clone_mock = Test::MockModule->new('OpenQA::Task::Git::Clone');
+        $clone_mock->redefine(_git_clone => sub { die "Internal API unreachable\n" });
+        subtest 'Internal API unreachable => skip' => sub {
+            my @gru_args = (
+                $t->app,
+                'git_clone',
+                {
+                    priority => 10,
+                    clones => 'git@gitlab.suse.de:qa/foo',
+                });
+            my $res;
+            combined_like {
+                $res = run_gru_job(@gru_args);
+            }
+            qr/Git server outage likely, skipping clone job/i,
+              'Minion clone job skipped due to short Git server outage';
+            is $res->{state}, 'finished', 'Job ended in "finished" state';
+            is $res->{result}, 'Job successfully executed', 'Job result indicates success (skip)';
+        };
+
+        subtest 'Internal API unreachable => fail' => sub {
+            $tmpdir->child('mock.other.flag')->touch;
+            my $old_time = time - 2000;    # "older" than 1800s
+            utime($old_time, $old_time, "$tmpdir/mock.other.flag");
+            my @gru_args = ($t->app, 'git_clone', {priority => 10});
```

Comment:
`priority` is a weird path, and `10` is a weird git url.
It works because of the fallback to `other`, if it is not http nor ssh, but we should not write confusing tests.